### PR TITLE
[14.0][MIG] report_py3o: Migration to 14.0

### PR DESCRIPTION
**Pull request by elmeriniemela**

### CODE REVIEW

**Comment by rvalyi** [comment]
Thank you for the work @elmeriniemela  , but please enforce the standard migration process here https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0 (that is mostly preserve the commit history). Please redo the pull request in the proper way, then we will be happy to review.

**Comment by elmeriniemela** [comment]
> Thank you for the work @elmeriniemela , but please enforce the standard migration process here https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0 (that is mostly preserve the commit history). Please redo the pull request in the proper way, then we will be happy to review.

@rvalyi I added the git history, thanks for the tip!

File: report_py3o/models/ir_actions_report.py
Comment (mourad-ehm) [inline report_py3o/models/ir_actions_report.py:181]:
```suggestion
            return safe_eval(report.print_report_name, {"object": obj,})
```
In v 14.0 time module is already in safe_eval https://github.com/odoo/odoo/blob/14.0/odoo/tools/safe_eval.py#L38

**Review by mourad-ehm — CHANGES_REQUESTED** [review]

File: report_py3o/models/ir_actions_report.py
Comment (hparfr) [reply report_py3o/models/ir_actions_report.py:181]: retreturn ?

Comment (mourad-ehm) [reply report_py3o/models/ir_actions_report.py:181]: fixed. thanks

**Review by hparfr — APPROVED** [review]
LGTM

**Comment by alexis-via** [comment]
It doesn't work when print_report_name is set. I fixed the bug in this commit: https://github.com/akretion/reporting-engine/commit/f10b5647ca7a37816fb8b3bc689e7bbcda867450
@elmeriniemela I wanted to make a PR to your branch, but I can't select it (maybe some special settings on your side ??). So please take the code in my commit and push it here.

**Comment by elmeriniemela** [comment]
Thanks @alexis-via ! I added your fix and made you a co-author. Also fixed some deprecation warnings.

**Review by alexis-via — APPROVED** [review]

**Comment by Asemtalat** [comment]
> _No description provided._

Hello 
i wand v14 from that medule 
**report_py3o**

**Comment by Asemtalat** [comment]
if anyone have thats medule (https://apps.openerp.com/apps/modules/13.0/report_py3o)    on v 14 plz send it to my mail (asemasem728@gmail.com)

**Comment by StefanRijnhart** [comment]
@elmeriniemela thank you for the port! Travis fails on the following error
```
FAIL: TestReportPy3o.test_py3o_report_availability

Traceback (most recent call last):

  File "/home/travis/build/OCA/reporting-engine/setup/report_py3o/odoo/addons/report_py3o/tests/test_report_py3o.py", line 228, in test_py3o_report_availability

    self.assertTrue(self.report.lo_bin_path)

AssertionError: False is not true
```
Please have a look and see if you can figure out what's wrong.

**Comment by etobella** [comment]
libreoffice is missing on the travis configuration. 

https://github.com/OCA/reporting-engine/blob/13.0/.travis.yml#L15

Can you add the needed libraries on .travis.yml?

**Comment by etobella** [comment]
Review precommit please.

Also, there is a warning related to genshi, but I don't know where it is generated
```
2021-02-10 14:46:43,478 160 WARNING openerp_test py.warnings: /.repo_requirements/virtualenv/python3.6/lib/python3.6/site-packages/genshi/template/interpolation.py:36: DeprecationWarning: Flags not at the start of the expression '[uU]?[rR]?("""|\\\'\\\'\\' (truncated)
PseudoToken
```

**Review by njeudy — APPROVED** [review]
Ok test, and using it in prod :)

**Comment by rbr-sacatec** [comment]
This is not working on windows. The following error message is shown when trying to print a py3o report:

> The application cannot be started. 
> The configuration file "C:\Program Files\LibreOffice\program\bootstrap.ini" is corrupt.

I've found this open issue on libreoffice-convert repository:
[https://github.com/elwerene/libreoffice-convert/issues/32](https://github.com/elwerene/libreoffice-convert/issues/32)

The error seems to be caused by the newly added `-env:UserInstallation=file` command line option, which currently shows an error message on windows.

I suggest this change on py3o_report.py line 272 to fix the error:
```py
if user_installation and os.name != 'nt':
    cmd.append("-env:UserInstallation=file:%s" % user_installation)
```

**Comment by elmeriniemela** [comment]
> Review precommit please.
> 
> Also, there is a warning related to genshi, but I don't know where it is generated
> 
> ```
> 2021-02-10 14:46:43,478 160 WARNING openerp_test py.warnings: /.repo_requirements/virtualenv/python3.6/lib/python3.6/site-packages/genshi/template/interpolation.py:36: DeprecationWarning: Flags not at the start of the expression '[uU]?[rR]?("""|\\\'\\\'\\' (truncated)
> PseudoToken
> ```

https://github.com/edgewall/genshi/issues/15

**Comment by alexis-via** [comment]
To be able to include properly formatted datetime field in a report displaying date and time, I propose this patch : https://people.via.ecp.fr/~alexis/py3o-format-datetime.diff
@elmeriniemela Could you include it? I would love to make a PR to your branch, but I can't select it for some reasons.

**Comment by alexis-via** [comment]
@elmeriniemela Could you make your branch public so that we can push a PR to your branch, to help you on this PR? Or do you prefer that we superseed this PR?
I'd like to push this PR to get it merged because I have 4 other OCA PR that are waiting for this one to be merged.

**Comment by elmeriniemela** [comment]
@alexis-via I'm not an expert on github settings and I could not find with ~15mins of searching whats the issue with my branch. If you know what I need to change I would be happy to make the branch public. If you don't know the root cause of this branch issue either, I could make you a collaborator of the fork?

**Comment by elmeriniemela** [comment]
@alexis-via I applied your patch. (just plain git apply and commit, I didnt test it myself)

**Review by alexis-via — APPROVED** [review]

File: report_py3o/models/py3o_report.py
Comment (nilshamerlinck) [inline report_py3o/models/py3o_report.py]:
```suggestion
    # workaround for https://github.com/edgewall/genshi/issues/15
    # that makes runbot build red because of the DeprecationWarning
    with warnings.catch_warnings():
        warnings.filterwarnings("ignore", category=DeprecationWarning)
        from py3o.template import Template
```

File: report_py3o/models/py3o_report.py
Comment (nilshamerlinck) [inline report_py3o/models/py3o_report.py:9]:
```suggestion
import tempfile
import warnings
```

File: report_py3o/models/py3o_report.py
Comment (nilshamerlinck) [inline report_py3o/models/py3o_report.py]:
```suggestion
            except OSError:
```

**Comment by nilshamerlinck** [comment]
Hi @elmeriniemela applying the suggested changes + running `pre-commit run -a` will make both checks green :)

**Comment by lmignon** [comment]
@elmeriniemela Thank you for all this work. Can you check the pre-commit issues https://github.com/OCA/reporting-engine/runs/4149149084?check_suite_focus=true#step:4:106 (mainly black problem).Once fixed I'll merge your PR

**Review by lmignon — CHANGES_REQUESTED** [review]
pre-commit issues to fix. Otherwise LGTM (code review only)

**Comment by leemannd** [comment]
Hello @elmeriniemela you can fix your PR by running `pre-commit run -a` or either merge the PR I did against your branch https://github.com/elmeriniemela/reporting-engine/pull/1

**Review by leemannd — APPROVED** [review]

**Comment by leemannd** [comment]
@OCA/reporting-engine-maintainers Ready to be merged :)

**Comment by leemannd** [comment]
Travis is failing because it is trying to fetch a closed PR with the same code. I'm not sure why it does so.

**Comment by nilshamerlinck** [comment]
Looks like a bug on travis side indeed. Given that the travis build was green before and pre-commit cleaning has now been done, I think this can be merged as is @lmignon @alexis-via :)

**Comment by lmignon** [comment]
@nilshamerlinck Thank you for all your work. I've restarted travis jobs. We need to have travis :green_circle: to be able to merge.

**Review by lmignon — APPROVED** [review]

**Comment by lmignon** [comment]
Strange... For travis, this pr seems to be linked to #551

**Pull request merged**
